File: 2021W04/fienestar/pgr_12982.js
/*
    예산
    https://programmers.co.kr/learn/courses/30/lessons/12982
*/

function solution(d, budget) {
    d.sort((a,b)=>a-b)
    for(let i in d)
        if(d[i] > budget)
            return +i
        else
            budget -= d[i]
    return d.length
}

File: 2022W27/brainer/pgr_12982.py
# 예산
# https://school.programmers.co.kr/learn/courses/30/lessons/12982

def solution(d, budget):
    if sorted(d)[0] > budget: return 0
    return next(len(sorted(d)[:i]) for i in range(len(d), 0, -1) if budget >= sum(sorted(d)[:i]))
